### PR TITLE
Sm trainer smp init fix

### PR DESCRIPTION
**Pull request by philschmid**
# What does this PR do?

Fixes `SageMakerTrainer` `smp.init` for `smp 1.3`. It also removes the `is_smdistributed_available` for are more robust `is_sagemaker_model_parallel_available`.

### CODE REVIEW

File: src/transformers/sagemaker/training_args_sm.py [hunk 1]
```diff
@@ -24,33 +26,48 @@
...
     import smdistributed.modelparallel.torch as smp
 
+    smp.init()
+
 
 @dataclass
 class SageMakerTrainingArguments(TrainingArguments):
-    mp_parameters: str = field(
-        default="", metadata={"help": "Used by the SageMaker launcher to send mp-specific args."}
-    )
```
Comment (sgugger) [inline src/transformers/sagemaker/training_args_sm.py:40]: Will this work with previous versions of `smdistributed` that were sending those parameters to the script?

**Review by sgugger — APPROVED** [review]
LGTM! I just have one question around the removal of one argument.

File: src/transformers/sagemaker/training_args_sm.py (hunk 1 above)
Comment (philschmid) [reply src/transformers/sagemaker/training_args_sm.py]: Added them again. SageMaker is passing `mp_parameters key=value,key=value` as args. https://github.com/huggingface/transformers/pull/10122/files#r574206511

**Comment by mansimane** [comment]
LGTM! I ran one training job successfully with these changes.

**Comment by philschmid** [comment]
I tested it with `pytorch1.7.1`
`564829616587.dkr.ecr.us-east-1.amazonaws.com/huggingface-pytorch-training:1.7.1-transformers4.4.0-py36-gpu-cu110-ubuntu18.04` 
and `pytorch1.6` 
`763104351884.dkr.ecr.us-east-1.amazonaws.com/huggingface-pytorch-training:1.6.0-transformers4.4.2-gpu-py36-cu110-ubuntu18.04`

**Pull request merged**